app/src/main/java/com/infomaniak/drive/ui/fileList/fileDetails/SelectCategoriesFragment.kt

```diff
@@ -22,134 +22,145 @@ import android.view.LayoutInflater
-            }
+            onCategoryChanged = { id, isSelected -> if (isSelected) addCategory(id) else removeCategory(id) }
         ).apply {
             this.canEditCategory = canEditCategory
             this.canDeleteCategory = canDeleteCategory
-
-            val uiCategories = DriveInfosController.getCurrentDriveCategories().map { category ->
-                val fileCategory = file.categories.find { it.id == category.id }
-                UICategory(
-                    id = category.id,
-                    name = category.getName(requireContext()),
-                    color = category.color,
-                    isPredefined = category.isPredefined,
-                    isSelected = fileCategory != null,
-                    userUsageCount = category.userUsageCount,
-                    addedToFileAt = fileCategory?.addedToFileAt,
-                )
-            }
-            setItems(uiCategories.sortCategoriesList())
-
-            onMenuClicked = { category ->
-                safeNavigate(
-                    R.id.categoryInfoActionsBottomSheetDialog, bundleOf(
-                        "fileId" to file.id,
-                        "categoryId" to category.id,
-                        "categoryName" to category.name,
-                        "categoryColor" to category.color,
-                        "categoryIsPredefined" to category.isPredefined,
-                    )
-                )
+            when (usageMode) {
+                SELECTED_CATEGORIES -> updateSelectedCategoriesModeUI()
+                else -> updateFileCategoriesModeUI()
             }
+            categoriesRecyclerView.adapter = this
         }
-
-        categoriesRecyclerView.adapter = categoriesAdapter
     }
 
     private fun setAddCategoryButton(canCreateCategory: Boolean) {
         toolbar.menu.findItem(R.id.addCategory).isVisible = canCreateCategory
     }
 
-    private fun setListeners() {
-        toolbar.apply {
-            setOnMenuItemClickListener { menuItem ->
-                if (menuItem.itemId == R.id.addCategory) {
-                    navigateToCreateCategory()
-                    true
-                } else false
-            }
-            setNavigationOnClickListener {
-                setBackNavigationResult(
-                    SELECT_CATEGORIES_NAV_KEY,
-                    categoriesAdapter.allCategories.filter { it.isSelected }.map { it.id },
-                )
+    private fun configureToolbar() = with(toolbar) {
+        setOnMenuItemClickListener { menuItem ->
+            if (menuItem.itemId == R.id.addCategory) {
+                navigateToCreateCategory()
+                true
+            } else {
+                false
             }
         }
+        setNavigationOnClickListener { setBackNavResult() }
+    }
 
-        createCategoryRow.setOnClickListener { navigateToCreateCategory() }
+    private fun configureSearchView() {
+        clearButton.setOnClickListener { searchView.text = null }
```

Comment:
I prefer `setText("")` instead of `text=null` 
```suggestion
        clearButton.setOnClickListener { searchView.setText("") }
```